nautobot/core/api/serializers.py

```diff
@@ -132,14 +180,60 @@ def get_field_names(self, declared_fields, info):
         fields = list(super().get_field_names(declared_fields, info))  # Meta.fields could be defined as a tuple
         self.extend_field_names(fields, "display", at_start=True)
         self.extend_field_names(fields, "id", at_start=True)
-        # Needed because we don't have a common base class for all nested serializers vs non-nested serializers
-        if not self.__class__.__name__.startswith("Nested"):
-            if hasattr(self.Meta.model, "created"):
-                self.extend_field_names(fields, "created")
-            if hasattr(self.Meta.model, "last_updated"):
-                self.extend_field_names(fields, "last_updated")
+        if hasattr(self.Meta.model, "created"):
+            self.extend_field_names(fields, "created")
+        if hasattr(self.Meta.model, "last_updated"):
+            self.extend_field_names(fields, "last_updated")
+        # This is here for the PolymorphicProxySerializers which
+        # are looking for an object_type field (originally on WritableNestedSerializer now BaseModelSerializer)
+        self.extend_field_names(fields, "object_type", at_start=True)
+        # Eliminate all field names that start with "_" as those fields are not user-facing
+        fields = [field for field in fields if not field.startswith("_")]
         return fields
 
+    def build_field(self, field_name, info, model_class, nested_depth):
+        """
+        Return a two tuple of (cls, kwargs) to build a serializer field with.
+        """
+        # For tags field, DRF does not recognize the relationship between tags and the model itself (?)
+        # so instead of calling build_nested_field() it will call build_property_field() which
+        # makes the field impervious to the `?depth` parameter.
+        # So we intercept it here to call build_nested_field()
+        # which will make the tags field be rendered with TagSerializer() and respect the `depth` parameter.
+        if (
+            field_name == "tags"
+            and isinstance(model_class.tags, (_NautobotTaggableManager, _TaggableManager))
```

Comment:
Probably just the isinstance check is sufficient here - I think even if we had a _TaggableManager field named something *other* than `tags` we'd still need this case?